packages/zwave-js/src/lib/node/Node.ts

```diff
@@ -2032,6 +2032,44 @@ protocol version:      ${this.protocolVersion}`;
 		return true;
 	}
 
+	/**
+	 * Interviews a command class unless the device configuration provides compat
+	 * values to use instead of performing the interview.
+	 */
+	private async compatInterviewCC(cc: CommandClass): Promise<void> {
+		const compatValues = this._deviceConfig?.compat?.interviewCCs
+			?.get(cc.ccId)
+			?.endpoints.get(cc.endpointIndex);
+		if (!compatValues) {
+			await cc.interview();
+		} else {
+			for (const compatValue of compatValues) {
+				if (cc.isInternalValue(compatValue.property as any)) {
+					const valueId: ValueID = {
+						commandClass: cc.ccId,
+						endpoint: cc.endpointIndex,
+						property: compatValue.property,
+						propertyKey: compatValue.propertyKey,
+					};
+					this.valueDB.setValue(valueId, compatValue.value);
+				} else {
+					this.driver.controllerLog.logNode(
+						this.id,
+						`failed to set compatibility value from device configuration for ${getCCName(
+							cc.ccId,
+						)}, endpoint ${cc.endpointIndex}: '${
+							compatValue.property
+						}' is not a registered internal value`,
+						"error",
+					);
+				}
+			}
+
+			await cc.refreshValues();
```

Comment:
We need to make this conditional. Probably by adding a method to the `CommandClass` class which defaults to `true` and can be overridden in subclasses. `UserCodeCC` for one does not call this during the interview.

Reply:
Hmm, interesting. The CCs all handle this behavior in slightly different ways. What are you thoughts on this?

1. We could add a `shouldRefreshValuesDuringCompatInterview()` method that is overridden by UserCodeCC and other CCs as needed to make the refresh conditional when using compatibility flags. This alternative would result in duplicated logic between the regular interview and compat interview code paths.
2. We could add a `shouldRefreshValuesDuringInterview()` method that is overridden by UserCodeCC and other CCs to make the refresh conditional during normal or compatibility interviews and then refactor places where those CCs directly call `refreshValues()` during the interview to call this new method (or perhaps a wrapper like `refreshValuesDuringInterviewIfNeeded()`). This alternative is a bit more invasive but it keeps the code paths more convergent.
3. We could let the compat interview flag itself state what should happen by way of a `refreshValues` option. This alternative requires the least changes but it risks breaking the expected behavior of `driver.options.interview.queryAllUserCodes` in UserCodeCC.

I'm leaning towards option 2 because I think it's a bit tidier in the long term.

Reply:
Me too, I'd probably use an overload for `refreshValues(partOfInterview: boolean)` which is marked `@internal` so it doesn't get exposed to users of the library. I initially thought this feature would require something like 2. anyways, but for every single query. Maybe I'll do that at some point anyways, but that is going to require larger changes to the value definitions first.

Your solution is much less invasive with the downside that the entire interview of a CC needs to be replicated in the config file.

Speaking of, we need to be able to set metadata during the fake interview as well. There are several devices which report incorrect color channels and I'd like to use this feature for the Color Switch CC too, which sets metadata:
https://github.com/zwave-js/node-zwave-js/blob/bb924723cf234c0c153cae74143e9cf8bf5b7170/packages/zwave-js/src/lib/commandclass/ColorSwitchCC.ts#L491-L511

This will be a bit ugly since the entire metadata object will require to be repeated in the config file, but I can live with that.

Reply:
Gotcha.  Adding metadata should be straightforward.

In preparation for some refactoring, I took a closer look at how `refreshValues()` gets called from `interview()` and I see a few different patterns in use. Can you check my assumptions?

1. Some CCs call `refreshValues()` unconditionally right at the end of the interview.
2. Some CCs call `refreshValues()` conditionally during the interview, such as `UserCodeCC` (based on a driver option) and `ThermostatSetpointCC` (version dependent, perhaps because with version <= 2 it's querying all of the setpoint values anyhow so it's kind of refreshing values as a side effect).
3. Some CCs perform additional work during the interview after calling `refreshValues()` that's important for using the device, such as `AssociationCC` (setting the lifeline) and `BasicCC` (removing the CC if it can't obtain the value).
4. Some CCs don't call `refreshValues()` during the interview at all, such as `WakeUpCC` (doesn't have any values).

So overall, I think we can classify the behavior into 4 cases:

1. CCs that don't have any values to refresh.
2. CCs that are refreshed with discretion (UserCodeCC). Is it just because it's slow or can there be undesirable side-effects too ?
3. CCs whose interview does not have a data dependency on values and only needs to ensure that they have been populated.
4. CCs whose interview has a data dependency on values because they are needed to determine the node's capabilities, to ensure that the CC actually seems to be working, or to configure it correctly.

My little hack skips the whole interview. Unfortunately, that skips all of its potential side-effects too.

I think a better solution might be to split the interview into three (or more?) steps:

1. `interview(driver)`: Query the node's capabilities and figure out what makes it tick and set up its internal values and metadata. This step might be altered or skipped due to compat flags.
2. `configure(driver)`: Configure the node to apply side-effects (such as setting lifeline associations).
3. `refreshValues(driver, true)`: Refresh any remaining values of the node that have yet to be queried since the interview started so that they're available to the user right away (except for cases like UserCodeCC where that's considered too expensive). We could always refresh all values at this step but that might result in redundant work in cases where the interview has data dependencies on certain values and will have queried them already unless we are using compat flags to skip the interview altogether.

Thoughts?

Reply:
I like it. I'm assuming `refreshValues` checks if the condition is fulfilled when it is indicated that it is called during an interview?